libcore/DndHandler.vala

```diff
@@ -294,42 +294,53 @@ namespace Files {
             return sb.str;
         }
 
-        // Used when dragging a file item
-        public static void set_selection_data_from_file_list (Gtk.SelectionData selection_data,
+        // Used when TargetType is TEXT_URI_LIST
+        public static void set_selection_data_as_file_list (Gtk.SelectionData selection_data,
                                                               GLib.List<Files.File> file_list,
                                                               string prefix = "") {
 
             GLib.StringBuilder sb = new GLib.StringBuilder (prefix);
-            set_stringbuilder_from_file_list (sb, file_list, true); // This will keep the "file://" protocol
+            set_stringbuilder_from_file_list (
+                sb,
+                file_list,
+                TargetType.TEXT_URI_LIST
+            );
             selection_data.@set (selection_data.get_target (),
                                  8,
                                  sb.data);
 
         }
 
-        // Used when copying a file item
-        public static void set_selection_text_from_file_list (Gtk.SelectionData selection_data,
+        // Used when TargetType is STRING
+        public static void set_selection_data_as_text (Gtk.SelectionData selection_data,
                                                               GLib.List<Files.File> file_list,
                                                               string prefix = "") {
 
             GLib.StringBuilder sb = new GLib.StringBuilder (prefix);
-            set_stringbuilder_from_file_list (sb, file_list, false); // This remove the "file://" protocol
+            set_stringbuilder_from_file_list (
+                sb,
+                file_list,
+                TargetType.STRING
+            );
             sb.truncate (sb.len - 2);  /* Do not want "\r\n" at end when pasting into text*/
             selection_data.set_text (sb.str, (int)(sb.len));
         }
 
         private static void set_stringbuilder_from_file_list (GLib.StringBuilder sb,
                                                               GLib.List<Files.File> file_list,
-                                                              bool keep_protocol) {
+                                                              uint info) {
```

Comment:
```suggestion
                                                              TargetType type) {
```

Reply:
Can we be sure that `info` is a valid TargetType?